src/providers/java.rs

```diff
@@ -0,0 +1,44 @@
+use super::Provider;
+use crate::nixpacks::{
+    app::App,
+    environment::Environment,
+    phase::{BuildPhase, InstallPhase, SetupPhase, StartPhase},
+};
+use anyhow::Result;
+
+pub struct JavaProvider {}
+
+impl Provider for JavaProvider {
+    fn name(&self) -> &str {
+        "Java"
+    }
+
+    fn detect(&self, app: &App, _env: &Environment) -> Result<bool> {
+        Ok(app.includes_file("pom.xml")
```

Comment:
I thought scala, groovy, clojure etc were going in different providers?

Reply:
WIP, will remove them later.

Reply:
Also, Gradle should probably be supported.

Reply:
I am basically rewriting [heroku's java buildpack](https://github.com/heroku/heroku-buildpack-java). Will add/remove stuff later.

Reply:
> I thought scala, groovy, clojure etc were going in different providers?

We're gonna do em as one for now and then work on composition once we get to parity

Reply:
May I ask why?

Reply:
Simplicity sake for now. Easier to break them up later once we know what the composition looks like than try to define abstractions across providers without knowing what should be composable.